analyses/stackanalyses/controller.go

```diff
@@ -0,0 +1,191 @@
+package stackanalyses
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/fabric8-analytics/cli-tools/analyses/driver"
+	"github.com/fabric8-analytics/cli-tools/analyses/pypi"
+	"github.com/fabric8-analytics/cli-tools/utils"
+	"github.com/jpillora/backoff"
+)
+
+// Controller is a control structure.
+type Controller struct {
+	// an implemented Matcher
+	m         driver.StackAnalysisInterface
+	fileStats *driver.ReadManifestResponse
+}
+
+// API Constants
+const (
+	APIStackAnalyses = "/api/v2/stack-analyses"
+)
+
+//StackAnalyses Performs Full Stack Analyses
+func StackAnalyses(requestParams driver.RequestType) driver.GetResponseType {
+	log.Info().Msgf("Performing full Stack Analyses. Please wait...")
+	log.Debug().Msgf("Executing StackAnalyses.")
+	b := &backoff.Backoff{
+		Min:    5 * time.Second,
+		Max:    120 * time.Second,
+		Factor: 2,
+		Jitter: false,
+	}
+	matcher, err := GetMatcher(requestParams.RawManifestFile)
+	if err != nil {
+		log.Fatal().Msgf(err.Error())
+	}
+	mc := NewController(matcher)
+	mc.fileStats = mc.buildFileStats(requestParams.RawManifestFile)
+	postResponse := mc.postRequest(requestParams, mc.fileStats.DepsTreePath)
+	getResponse := mc.getRequest(requestParams, postResponse, b)
```

Comment:
This should this be under polling loop. similar to vscode extension.

Reply:
Polling is taken care by Backoff https://github.com/fabric8-analytics/cli-tools/blob/6ef0892cea4f8030c06324b356491a4b7d10703f/analyses/stackanalyses/controller.go#L38-L43 

More on https://github.com/jpillora/backoff